iam.tf

```
@@ -22,12 +22,11 @@ data "aws_iam_policy_document" "lambda_basic" {
     effect = "Allow"
 
     actions = [
-      "logs:CreateLogGroup",
       "logs:CreateLogStream",
       "logs:PutLogEvents",
     ]
 
-    resources = ["arn:aws:logs:*:*:*"]
+    resources = ["${aws_cloudwatch_log_group.lambda.arn}"]
```

I think these permissions need a slight update, based on [the docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/iam-access-control-overview-cwl.html) the ARN format for log streams would be: `arn:aws:logs:region:account-id:log-group:log_group_name:log-stream:log-stream-name`

```suggestion
    resources = ["${aws_cloudwatch_log_group.lambda.arn}:log-stream:*"]
```

---

I investigated this a bit and found that when referencing the `arn` of `aws_cloudwatch_log_group` Terraform injects a wildcard already:

```
$ terraform version
Terraform v0.12.7
+ provider.aws v2.31.0
```

```
resource "aws_cloudwatch_log_group" "lambda" {
    arn                         = "arn:aws:logs:us-west-2:0000000000:log-group:/aws/lambda/notify_slack:*"
    id                           = "/aws/lambda/notify_slack"
    name                     = "/aws/lambda/notify_slack"
    retention_in_days = 30
    tags                       = {}
}
```

It still works (the lambda can still log to new log streams) even when appending `:log-stream:*`. Not sure which syntax would be better to use....